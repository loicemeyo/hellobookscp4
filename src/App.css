.logo {
  height: 150px;
}
.subtitle {
  font-style: italic;
  font-size: x-large;
  color: #3e15ad;
  text-align: center;
}
.background {
/* background-image: url('./assets/library2.jpg'); */
background-color: bisque;
background-position: center;
background-size: cover;
width: 100%;
height: 100%;
text-align: center;
color: white;
font-family: 'Courier New', Courier, monospace;
font-weight: 1200;
}
.homecontent {
  background-image: url ('/assets/hellobooks.jpg');
  background-repeat: no-repeat;
  height: 80%;
  width: 100%;
  position: relative;
  background-size: cover !important;
  margin-left: 10px !important;
  margin-right: 10px !important;
  margin-top: 20px !important;
  margin-bottom: 10px !important;
}
.inline-buttons {
  margin-left: 15px;
  margin-right: 15px;
  display: table;
  margin: 0 auto;
  width: 100%;
  margin: 20px;
  text-align: center;
}
.outer
{
  margin-left: 20px;
  width:100%;
  text-align: center;
}
.navbar-nav {
  float: right !important;
}
.nav-item {
  padding : 0 25px 0 25px;
}
#allbooks {
  width:95%;
  padding: 10px;
  margin: auto;
}
#onebook {
  background-color: #eee;
  border-radius: 5px;
  box-shadow: 5px 10px 0 #888888;
  margin: 10px;
}
#bookinfo{
  text-align: center;
}
#singlebook {
  box-sizing: content-box;    
  width: 400px;
  height: 350px;
  padding: 10px; 
  border-radius: 5px;
  box-shadow: 10px 15px 0 #888888;
  margin: 10px;
  background-color: #eee;
  font-size: 16px;
  text-align: center;
  margin-left: 300px;
  margin-top: 20px
}
.inner
{
  display: inline-block;
  margin-left: 10px;
  margin-right: 10px;
  
}

.footer {
    bottom: 0;
    width: 100%;
    position: fixed;
    height: 60px;
    background-color:#337ab7!important;
    margin-bottom: 10px;
    text-align: center;
    color: aliceblue;
}
.footer-block {
      margin: 20px 0;
    }
#input{
  text-align: center;
}
#welcomePage {
  width: 50%;
  margin-left: 24%;
  margin-top: 16%;
}
#signupPage {
  width: 50%;
  margin-left: 24%;
  margin-top: 5%;
}
#loginPage{
  width: 50%;
  margin-left: 24%;
  margin-top: 10%;
}
#login{
  width:100%;
}