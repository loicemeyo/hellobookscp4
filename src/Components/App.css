.logo {
  height: 150px;
}
.subtitle {
  font-style: italic;
  font-size: x-large;
  color: #3e15ad;
  text-align: center;
}
.homecontent {
  background-image: url ('/assets/hellobooks.jpg');
  background-repeat: no-repeat;
  height: 80%;
  width: 100%;
  position: relative;
  background-size: cover !important;
  margin-left: 10px !important;
  margin-right: 10px !important;
  margin-top: 20px !important;
  margin-bottom: 10px !important;
}
.inline-buttons {
  margin-left: 15px;
  margin-right: 15px;
  display: table;
  margin: 0 auto;
  width: 100%;
  margin: 20px;
  text-align: center;
}
.outer
{
  width:100%;
  text-align: center;
}
.inner
{
  display: inline-block;
  margin-left: 10px;
  margin-right: 10px
}
.form-inline {
  
}