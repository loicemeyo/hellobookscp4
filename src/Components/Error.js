import React, {Component} from "react";
import error from "../assets/uh-oh.jpg";

/**
 * This component handles invalid page requests
 */
class Error extends Component {
  render () {
    return (
      <div>
        <div id="error">
          <img alt="Invaid Page" src={error}/>
        </div>
        <br/>
        <div id="error2">
          <p> Sorry! This page does not exist </p>
        </div>
      </div>
    );

  }
    
}
export default Error;
