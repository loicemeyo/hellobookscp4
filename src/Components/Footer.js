import React from "react";

/**
 * This component displays the footer
 */

class Footer extends React.Component {
  render() {
    return (
      <div className="footer">
        <div class="container">
          <p class="footer-a">HelloBooks &copy; 2018</p>
          <p class="footer-block"> Contact <a href={"mailto:loicemeyo@gmail.com"} style={{color:'aliceblue'}}> Loice Meyo </a></p>
          <p class="footer-block">  Designed by Odi </p>
    
        </div>
      </div>
    );
  }
}
export default Footer;