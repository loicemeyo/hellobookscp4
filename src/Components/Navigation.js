import React from "react";
import { Link, browserHistory} from "react-router";
import axios from 'axios';
import swal from 'sweetalert';

const Base_url = "https://hellobooks-api-db.herokuapp.com"
/**
 * Navigation Component
 * Displays signup/login when user is not logged in
 * Displays logout upon login
 * The Home tab redirects a normal user to the view all books but redirects admin to view admin page
 */

class Navigation extends React.Component {

  /**
   * The logout function is embedded within navigation
   */
  handleLogout = (event) => {
    const loggingoutuser = {

        email: localStorage.getItem('email'),

    };
    const token = localStorage.getItem('access_token');
    const config ={ headers:{"Authorization":"Bearer " + token}}

    axios.post(`${Base_url}/api/v2/auth/logout`, loggingoutuser, config)
        .then(response => {
            localStorage.removeItem("access_token")
            localStorage.removeItem('email')
            browserHistory.push('/')
            swal("You have successfully logged out");
         
        }).catch(error => {
          console.log(error)
              if (error.response.status === 401) {
                localStorage.removeItem('access_token');
                localStorage.removeItem('email')
                browserHistory.push('/login');
             
          }
        });


};
  render() {
    let isloggedIn = localStorage.getItem("access_token");
    let isadmin=JSON.parse(localStorage.getItem("admin"));

    return (
      
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <a className="navbar-brand"><Link to="/">Hello Books</Link></a>
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>
        {isloggedIn ?(
          <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav">
          {isadmin? (
            <li className="nav-item active">
            <a className="nav-link"><Link to="/admin" className="inner">Home</Link>
            </a>
          </li>):
          (
            <li className="nav-item active">
            <a className="nav-link"><Link to="/books" className="inner">Home</Link>
            </a>
          </li>
          )}
            <li className="nav-item active">
              <a className="nav-link"><Link to="/users/books" className="inner">User History</Link>
              </a>
            </li>
            <li className="nav-item active">
              <a className="nav-link"><Link id="logout" style={{ cursor: 'pointer'}} onClick={this.handleLogout} className="inner">Logout</Link>
              </a>
            </li>
          </ul>
          
          </div>
        ):(
          <div className="collapse navbar-collapse" id="navbarNav">
            <ul className="navbar-nav">
              <li className="nav-item active">
                <a className="nav-link"><Link to="/" className="inner">Home</Link>
                </a>
              </li>
              <li className="nav-item active">
                <a className="nav-link"><Link to="/signup" className="inner">Signup</Link>
                </a>
              </li>
              <li className="nav-item active">
                <a className="nav-link"><Link to="/login" className="inner">Login</Link>
                </a>
              </li>
            </ul>
          </div>
        )}
      </nav>
    );
  }
}
export default Navigation;
export { Base_url }