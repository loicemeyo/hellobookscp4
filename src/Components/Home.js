import React, { Component } from "react";
import "../App.css";
import { Link } from "react-router";
import "react-bootstrap";
/**
 * The Home component displays the welcome message and signup/login options
 */

class Home extends Component {
  render() {
    return (
      <div>
        <div className="jumbotron" id="welcomePage">
          <p>Hello-Books is a simple application that helps manage a library <br />
						and its processes like stocking, tracking and renting books.
          </p>
          <div id="outer">
            <Link to="/signup" class="inner"><button type="submit" class="btn btn-primary">Get Started</button></Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Home;
