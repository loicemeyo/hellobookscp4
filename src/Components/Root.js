import React, { Component } from "react";
import Navigation from "./navbar";
import Footer from "./Footer";

class Root extends Component {
  // const tokenvalid=
  render() {
    return (
      <div>
        <Navigation/>
        {this.props.children}
        <Footer />
      </div>
    );
  }
}

export default Root;
