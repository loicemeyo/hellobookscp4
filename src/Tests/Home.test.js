import React from "react";
import { shallow, mount } from "enzyme";
import shallowToJson, { mountToJson } from "enzyme-to-json";
import Home from "../Components/Home";
import Root from "../Components/Root";
import Navigation from "../Components/Navigation";
import Error from "../Components/Error";
import Loginform from "../forms/login";
import Signupform from "../forms/signup";
import Footer from "../Components/Footer";
import Requestform from "../forms/requestreset";
import ReactDOM from "react-dom";
import App from "../App";
import moxios from "moxios";
import sinon from "sinon";


xit("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("Test Home Component", () => {
  const wrapper = shallow(<Home />);

  it("Home component renders without crushing", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});

describe("Test Root Component", () => {
  const wrapper = shallow(<Root />);

  it("Root component renders without crushing", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
describe("Test Navigation Component", () => {
  const wrapper = shallow(<Navigation />);

  it("Navigation component renders without crushing", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
describe("Test Login Component", () => {
  beforeEach(() => {
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
  });
  it("Login component renders without crushing", () => {
    let wrapper = shallow(<Loginform />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
  it("Calls handleSubmit() on loginform submission", () => {
    let wrapper = shallow(<Loginform />);
    let handleSubmit = sinon.spy();
    wrapper = mount(<Loginform onSubmit={handleSubmit} />);
    wrapper.find("form").simulate("submit");

    moxios.wait(() => { });
  });
  it("Calls handleChange() on inputting email field", () => {
    let wrapper = shallow(<Loginform />);
    let handleChange = sinon.spy();
    wrapper = mount(<Loginform onChange={handleChange} />);
    wrapper.find("#email").simulate("change");
  });
  it("Calls handleChange() on inputting password field", () => {
    let wrapper = shallow(<Loginform />);
    let handleChange = sinon.spy();
    wrapper=mount(<Loginform onChange={handleChange} />);
    wrapper.find("#password").simulate("change");
  });

});
describe("Test Signup Component", () => {
  beforeEach(() => {
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
  });

  it("Signup component renders without crushing", () => {
    let wrapper = shallow(<Signupform />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
  it("Calls handleSubmit() on signup form submission", () => {
    let wrapper = shallow(<Signupform />);
    let handleSubmit = sinon.spy();
    wrapper = mount(<Signupform onSubmit={handleSubmit} />);
    wrapper.find("form").simulate("submit");

    moxios.wait(() => { });

  });
  it("Calls handleChange() on inputting username field", () => {
    let wrapper = shallow(<Signupform />);
    let handleChange = sinon.spy();
    wrapper = mount(<Signupform onChange={handleChange} />);
    wrapper.find("#username").simulate("change");

  });
  it("Calls handleChange() on inputting email field", () => {
    let wrapper = shallow(<Signupform />);
    let handleChange = sinon.spy();
    wrapper = mount(<Signupform onChange={handleChange} />);
    wrapper.find("#email").simulate("change");

  });
  it("Calls handleChange() on inputting password field", () => {
    let wrapper = shallow(<Signupform />);
    let handleChange = sinon.spy();
    wrapper = mount(<Signupform onChange={handleChange} />);
    wrapper.find("#password").simulate("change");

  });
  it("Calls handleChange() on inputting confirm password field", () =>{
    let wrapper = shallow(<Signupform />);
    let handleChange = sinon.spy();
    wrapper = mount(<Signupform onChange={handleChange} />) ;
    wrapper.find("#passwordb").simulate("change");
  });


});
describe("Test Footer Component", () => {
  const wrapper = shallow(<Footer />);

  it("Footer component renders without crushing", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
describe("Test Requestform Component", () => {
  const wrapper = shallow(<Requestform />);

  it("Request password component renders without crushing", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
describe("Test Error Component", () => {
  const wrapper = shallow(<Error />);

  it("Error component renders without crushing", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});