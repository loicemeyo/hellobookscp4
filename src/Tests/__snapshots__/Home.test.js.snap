// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Error Component Error component renders without crushing 1`] = `
<div>
  <div
    id="error"
  >
    <img
      src="uh-oh.jpg"
    />
  </div>
  <br />
  <div
    id="error2"
  >
    <p>
       Sorry! This page does not exist 
    </p>
  </div>
</div>
`;

exports[`Test Footer Component Footer component renders without crushing 1`] = `
<div
  className="footer"
>
  <div
    class="container"
  >
    <p
      class="footer-block"
    >
       Copyright 2018. Loice Meyo 
    </p>
  </div>
</div>
`;

exports[`Test Home Component Home component renders without crushing 1`] = `
<div>
  <div
    className="jumbotron"
    id="welcomePage"
  >
    <p>
      Hello-Books is a simple application that helps manage a library 
      <br />
      and its processes like stocking, tracking and renting books.
    </p>
    <div
      id="outer"
    >
      <Link
        class="inner"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/signup"
      >
        <button
          class="btn btn-primary"
          type="submit"
        >
          Get Started
        </button>
      </Link>
    </div>
  </div>
</div>
`;

exports[`Test Login Component Login component renders without crushing 1`] = `
<div
  className="container"
  id="loginPage"
>
  <div
    className="jumbotron"
  >
    <form
      onSubmit={[Function]}
    >
      <h2>
         Log in 
      </h2>
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            defaultValue=""
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            required={false}
            type="password"
          />
        </div>
      </div>
      <br />
      <button
        className="btn btn-default"
        type="submit"
      >
        Login
      </button>
      <br />
      <br />
      <div>
        Forgot password? 
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/requestreset"
        >
          Click here to reset your password
        </Link>
      </div>
    </form>
  </div>
</div>
`;

exports[`Test Navigation Component Navigation component renders without crushing 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <a
    className="navbar-brand"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/"
    >
      Hello Books
    </Link>
  </a>
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarNav"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarNav"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
        >
          <Link
            className="inner"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            Home
          </Link>
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
        >
          <Link
            className="inner"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/signup"
          >
            Signup
          </Link>
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
        >
          <Link
            className="inner"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/login"
          >
            Login
          </Link>
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Test Root Component Root component renders without crushing 1`] = `
<div>
  <Navigation />
  <Footer />
</div>
`;

exports[`Test Signup Component Signup component renders without crushing 1`] = `
<div>
  <div
    className="jumbotron"
    id="signupPage"
  >
    <form
      onSubmit={[Function]}
    >
      <h2>
         Sign up 
      </h2>
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="username"
            name="name"
            onChange={[Function]}
            placeholder="Enter Username"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Enter Email"
            required={true}
            type="email"
            value=""
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            defaultValue=""
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            required={false}
            type="password"
          />
        </div>
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-xs-6"
        >
          <input
            className="form-control"
            defaultValue=""
            id="passwordb"
            name="passwordb"
            onChange={[Function]}
            placeholder="Confirm Password"
            required={false}
            type="password"
          />
        </div>
      </div>
      <br />
      <button
        className="btn btn-default"
        type="submit"
      >
        Signup
      </button>
    </form>
  </div>
</div>
`;
